siriuspy/siriuspy/devices/ids.py

```diff
@@ -834,6 +864,122 @@ def cmd_clear_error(self):
+        'PolModeAcc-SP', 'PolModeAcc-RB',
+        'GainModeVelo-SP', 'GainModeVelo-RB',
+        'GainModeAcc-SP', 'GainModeAcc-RB',
+        )
+
+    def __init__(self, devname=None, props2init='all', auto_monitor_mon=True):
+        """."""
+        # check if device exists
+        if devname is None:
+            devname = self.DEVICES.DELTA52_10SB
+        if devname not in self.DEVICES.ALL:
+            raise NotImplementedError(devname)
+
+        # call base class constructor
+        super().__init__(
+            devname, props2init=props2init, auto_monitor_mon=auto_monitor_mon)
+
+    # --- pparameter ---
+
+    @property
+    def pparameter_speed_max(self):
+        """Return max pparameter speed readback [mm/s]."""
+        return self['MaxVelo-RB']
+
+    @property
+    def pparameter_speed_max_lims(self):
+        """Return max pparameter speed limits."""
+        ctrl = self.pv_ctrlvars('MaxVelo-SP')
+        lims = [ctrl['lower_ctrl_limit'], ctrl['upper_ctrl_limit']]
+        return lims
+
+    @property
+    def pparameter_speed(self):
+        """Return pparameter speed readback [mm/s]."""
+        return self['PolModeVelo-RB']
+
+    @property
+    def pparameter_min(self):
+        """Return ID pparameter lower control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['lower_ctrl_limit']
+
+    @property
+    def pparameter_max(self):
+        """Return ID pparameter upper control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('PolShift-Mon')
+        return ctrlvars['upper_ctrl_limit']
+
+    @property
+    def pparameter_mon(self):
+        """Return ID pparameter monitor [mm]."""
+        return self['PolShift-Mon']
+    
+    # --- kparameter ---
+
+    @property
+    def kparameter_speed_max(self):
+        """Return max kparameter speed readback [mm/s]."""
+        return self['MaxVelo-RB']
+
+    @property
+    def kparameter_speed_max_lims(self):
+        """Return max kparameter speed limits."""
+        ctrl = self.pv_ctrlvars('MaxVelo-SP')
+        lims = [ctrl['lower_ctrl_limit'], ctrl['upper_ctrl_limit']]
+        return lims
+
+    @property
+    def kparameter_speed(self):
+        """Return kparameter speed readback [mm/s]."""
+        return self['GainModeVelo-RB']
+
+    @property
+    def kparameter(self):
+        """Return ID kparameter readback [mm]."""
+        return self['Shift-RB']
+    
+    @kparameter.setter
+    def kparameter(self, value):
+        """Set ID kparameter [mm]."""
+        self['Shift-SP'] = value
+
+    @property
+    def kparameter_min(self):
+        """Return ID kparameter lower control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('Shift-SP')
+        return ctrlvars['lower_ctrl_limit']
+
+    @property
+    def kparameter_max(self):
+        """Return ID kparameter upper control limit [mm]."""
+        ctrlvars = self.pv_ctrlvars('Shift-SP')
+        return ctrlvars['upper_ctrl_limit']
```

Comment:
I don't see the properties 'Shift-RB' and 'Shift-SP' in the PROPERTIES_DEFAULT of DELTA device nor the ID device from which DELTA is derived. Aren't the properties 'GainShift-RB' and 'GainShift-SP'?

Reply:
deprecated